src/vmm/src/vstate/memory.rs

```diff
@@ -309,12 +326,11 @@ impl GuestMemoryExtension for GuestMemoryMmap {
                 let mut write_size = 0;
                 let mut dirty_batch_start: u64 = 0;
 
-                for (i, v) in kvm_bitmap.iter().enumerate() {
+                for i in 0..kvm_bitmap.len() {
```

Comment:
Now that we no longer need to do the bit shifting to read the kvm bitmap, can we simplify this loop to be something like `for page_offset in (0..region.size()).step_by(page_size)`?

On the other hand, even though snapshot creation is not a hotpath, maybe it's nice not always pay the penalty of copying the KVM log to the Firecracker log. Snapshot creation is very unlikely to fail, and copying the log involves a ton of atomic operations, so ideally, we would only copy the KVM log to the Firecracker log if something actually went wrong. So instead of doing

```
1. copy
2. iterate over firecracker log and write snapshot
```

we can do 

```
try {
   iterate over firecracker and kvm log and write snapshot
} catch {
   copy kvm log to firecracker log
} else {
   clear firecracker log
}
```

@JonathanWoollett-Light @pb8o what do you think?